### PR TITLE
[FG:InPlacePodVerticalScaling] Show CPU request in status only when configured

### LINKED ISSUE #126195: [FG:InPlacePodVerticalScaling] Minimum CPU request is displayed when only memory request is configured
### What happened?

If `InPlacePodVerticalScaling` is enabled and only a memory resource request is configured in a pod, the minimum CPU request set internally is displayed in the pod status:
```
$ kubectl get pod testpod -o jsonpath='spec: {.spec.containers[0].resources}{"\nallocatedResources: "}{.status.containerStatuses[0].allocatedResources}{"\nstatus: "}{.status.containerStatuses[0].resources}{"\n"}'
spec: {"requests":{"memory":"200Mi"}}
allocatedResources: {"memory":"200Mi"}
status: {"requests":{"cpu":"2m","memory":"200Mi"}}
```

### What did you expect to happen?

The minimum CPU request is not displayed in the pod status.

### How can we reproduce it (as minimally and precisely as possible)?

1. Enables the `InPlacePodVerticalScaling` feature.
2. Create a pod where only memory request is configured:
   <Details>

   ```
   apiVersion: v1
   kind: Pod
   metadata:
     labels:
       run: testpod
     name: testpod
   spec:
     containers:
     - image: nginx
       name: testpod
       resources:
         requests:
           memory: 200Mi
     restartPolicy: Always
   ```

   </Details>
3. See its resources in the pod status:
   ```
   $ kubectl get pod testpod -o jsonpath='spec: {.spec.containers[0].resources}{"\nallocatedResources: "}{.status.containerStatuses[0].allocatedResources}{"\nstatus: "}{.status.containerStatuses[0].resources}{"\n"}'
   spec: {"requests":{"memory":"200Mi"}}
   allocatedResources: {"memory":"200Mi"}
   status: {"requests":{"cpu":"2m","memory":"200Mi"}}
   ```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.30.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.0
```

</details>

### Cloud provider

N/A

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by hshiina**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
The minimum CPU request is set to a container internally if a CPU request is not configured. This minimum request does not need displaying to users. This PR sets a CPU request to a pod status only when a CPU request is configured in the pod spec.

This PR also prevents the internal minimum CPU request from triggering an unexpected resize in `computePodResizeAction()`.

#### Which issue(s) this PR fixes:

Fixes #126195

#### Does this PR introduce a user-facing change?

```release-note
Fixed a bug that the internal minimum CPU request is displayed in a pod status when only a memory request is configured in the pod and the `InPlacePodVerticalScaling` feature is enabled.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/pod_resize_test.go [hunk 1]
```diff
@@ -1131,6 +1131,24 @@ func doPodResizeTests() {
...
+			patchString: `{"spec":{"containers":[
+						{"name":"c1", "resources":{"requests":{"memory":"300Mi"}}}
+					]}}`,
+			expected: []TestContainerInfo{
+				{
+					Name:      "c1",
+					Resources: &ContainerResources{MemReq: "300Mi"},
+				},
+			},
+		},
```
Comment (bart0sh) [inline test/e2e_node/pod_resize_test.go:1151]: How about testing "CPU only" requests as well?

Comment (hshiina) [reply test/e2e_node/pod_resize_test.go:1151]: Thank you for the suggestion. I added a similar test case for CPU. In this case, CPU limit is necessary due to the issue #126388.

**Comment by hshiina** [comment]
It looks hitting a flake #126392.

**Comment by tallclair** [comment]
I think this PR is superseded by https://github.com/kubernetes/kubernetes/pull/128680

Please take a look at the linked PR.

/close

**Pull request closed without merging**